src/plugins/auctionHouseModule/Bid.ts

```diff
@@ -0,0 +1,104 @@
+import { PublicKey } from '@solana/web3.js';
+import {
+  amount,
+  BigNumber,
+  DateTime,
+  lamports,
+  Pda,
+  SolAmount,
+  SplTokenAmount,
+  toBigNumber,
+  toDateTime,
+  toOptionDateTime,
+} from '@/types';
+import { BidReceiptAccount } from './accounts';
+import { TokenWithMetadata } from '../nftModule';
+import { assert, Option } from '@/utils';
+import { AuctionHouse } from './AuctionHouse';
+
+export type Bid = Readonly<{
+  model: 'bid';
+  lazy: false;
+
+  // Models.
+  auctionHouse: AuctionHouse;
+  token: TokenWithMetadata;
+
+  // Addresses.
+  tradeStateAddress: Pda;
+  buyerAddress: PublicKey;
+  bookkeeperAddress: Option<PublicKey>;
+  receiptAddress: Option<Pda>;
+  purchaseReceiptAddress: Option<PublicKey>;
+
+  // Data.
+  price: SolAmount | SplTokenAmount;
+  tokens: SplTokenAmount;
+  createdAt: DateTime;
+  canceledAt: Option<DateTime>;
+}>;
+
+export const isBid = (value: any): value is Bid =>
+  typeof value === 'object' && value.model === 'bid' && !value.lazy;
+
+export function assertBid(value: any): asserts value is Bid {
+  assert(isBid(value), `Expected Bid type`);
+}
+export const toBid = (
+  account: BidReceiptAccount,
+  auctionHouseModel: AuctionHouse,
+  tokenModel: TokenWithMetadata
+): Bid => {
+  const lazyBid = toLazyBid(account, auctionHouseModel);
+  return {
+    ...lazyBid,
+    model: 'bid',
+    lazy: false,
+    token: tokenModel,
+    tokens: amount(lazyBid.tokens, tokenModel.mint.currency),
+  };
+};
+
+export type LazyBid = Omit<Bid, 'lazy' | 'token' | 'tokens'> &
+  Readonly<{
+    lazy: true;
+    metadataAddress: PublicKey;
+    tokens: BigNumber;
+  }>;
```

Comment:
You're forgetting another important piece of information provided by the `BidReceiptAccount` which is not provided by the `ListingReceiptAccount`:

```ts
tokenAccount: Option<PublicKey>;
```

Therefore, if we include my comment above, we should end up with the following type.

```suggestion
export type LazyBid = Omit<Bid, 'lazy' | 'token' | 'mint' | 'tokens'> &
  Readonly<{
    lazy: true;
    metadataAddress: PublicKey;
    tokenAddress: Option<PublicKey>; // <- This is important, if it's null, we know it's a public bid.
    tokens: BigNumber;
  }>;
```